src/datastore/postgres-store.ts

```diff
@@ -4310,6 +4373,156 @@ export class PgDataStore
     return events;
   }
 
+  /**
+   * It retrieves filtered events from the db based on transaction, principal or event type. Note: It does not accept both principal and txId at the same time
+   * @param args - addressOrTxId: filter for either transaction id or address
+   * @param args - eventTypeFilter: filter based on event types ids
+   * @param args - limit: returned that many rows
+   * @param args - offset: skip that any rows
+   * @returns returns array of events
+   */
```

Comment:
Looks good, but move it to `common.ts` so the comment appears in tooltips everywhere